recipes/folly/all/conanfile.py

```diff
@@ -44,44 +45,32 @@ def config_options(self):
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
 
         if self.settings.os == "Windows" and self.settings.arch != "x86_64":
-            raise ConanInvalidConfiguration("Folly requires a 64bit target architecture")
-        elif self.settings.os == "Windows" and self.settings.compiler == "Visual Studio" and \
-                "MT" in self.settings.compiler.runtime:
-            raise ConanInvalidConfiguration("Folly could not be build with runtime MT")
-        elif self.settings.os == "Macos" and self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built by apple-clang as shared library")
-        elif self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on Windows as a shared library")
-        elif Version(self.version) >= "2020.08.10.00" and self.settings.compiler == "Visual Studio" and \
-                not self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on Windows as a static library")
-        elif Version(self.version) >= "2020.08.10.00" and self.settings.compiler == "clang" and \
-                self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built by clang as a shared library")
+            raise ConanInvalidConfiguration("Folly requires a 64bit target architecture on Windows")
 
-        self._strip_options_requirements()
+        if self.settings.os in ["Macos", "Windows"] and self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built on {} as shared library".format(self.settings.os))
 
-    def _strip_options_requirements(self):
-        self.options["boost"].header_only = False
-        for boost_comp in self._required_boost_components:
-            setattr(self.options["boost"], "without_{}".format(boost_comp), False)
+        if self.version == "2020.08.10.00" and self.settings.compiler == "clang" and self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built by clang as a shared library")
 
-    @property
-    def _required_boost_components(self):
-        return ["context", "filesystem", "program_options", "regex", "system", "thread"]
+    def validate(self):
+        if self.options["boost"].header_only:
+            raise ConanInvalidConfiguration("Folly could not be built with a header only Boost")
+
+        for boost_comp in ["context", "filesystem", "program_options", "regex", "system", "thread"]:
+            if getattr(self.options["boost"], "without_{}".format(boost_comp), False):
+                raise ConanInvalidConfiguration("Folly could not be built with Boost without {}".format(boost_comp))
```

Comment:
I prefer the previous version. It displays all required options, so that consumers can quickly their boost options.

Reply:
Also please move `validate()` after `requirements()`

Reply:
Have to fix the merge conflict so i'll revert that 👍